packages/zambdas/src/ehr/create-update-medication-order/index.ts

```diff
@@ -115,31 +138,31 @@ async function updateOrder(
     newMedicationCopy = createMedicationCopy(existingMedicationCopy, orderData);
   }
 
-  const resultPromises: Promise<any>[] = [];
+  const transactionRequests: BatchInputRequest<FhirResource>[] = [];
   console.log(
     `Updating MedicationAdministration, orderData present: ${Boolean(orderData)}, newMedicationCopy present: ${Boolean(
       newMedicationCopy
     )}, newStatus: ${newStatus}`
   );
-
-  let updatedMedicationResource: MedicationAdministration = orderResources.medicationAdministration;
-
+  let updatedMedicationAdministration: MedicationAdministration | undefined = undefined;
+  const medicationAdministrationPatchOperations: Operation[] = [];
   if (orderData && newMedicationCopy) {
-    updatedMedicationResource = await updateMedicationAdministrationData({
-      oystehr,
+    updatedMedicationAdministration = updateMedicationAdministrationData({
       orderData,
       orderResources,
       administeredProviderId: newStatus !== undefined ? practitionerIdCalledZambda : undefined,
       medicationResource: newMedicationCopy,
     });
-    console.log('MedicationAdministration data was successfully updated.');
   }
 
   if (currentStatus && newStatus) {
-    resultPromises.push(changeOrderStatus(oystehr, orderResources, newStatus));
-    if (newStatus === 'administered') {
-      console.log('Creating MedicationStatement resource on administered action');
+    if (updatedMedicationAdministration != null) {
```

Comment:
`false` will pass it. Maybe consider `if (updatedMedicationAdministration) {` - it's more reliable.